### PR TITLE
Fix deprecation with isMethodSafe

### PR DESCRIPTION
**Pull request by jderusse**

### CODE REVIEW

File: src/Http/RequestMatcher/UnsafeRequestMatcher.php [hunk 1]
```diff
@@ -18,6 +18,14 @@ class UnsafeRequestMatcher implements RequestMatcherInterface
 {
     public function matches(Request $request)
     {
-        return !$request->isMethodSafe(false);
+        // hack needed for compatibility with SF 3.4 => 4.3
+        // sf 3.4 => isMethodSafe(false)
+        // sf 4.3 => isMethodSafe() or isMethodSafe(false)
+        // sf 4.4 => isMethodSafe()
+        if (method_exists($request, 'getPreferredFormat')) {
```
Comment (jderusse) [inline src/Http/RequestMatcher/UnsafeRequestMatcher.php]: Do you prefer using `Kernel::VERSION_ID` ? given this method `getPreferredFormat` is not related to the tested behavior?

Comment (dbu) [reply src/Http/RequestMatcher/UnsafeRequestMatcher.php]: that actually would be more explict. you could do `if (version_compare(Kernel::VERSION, '4.3.0', '>=')) {`

**Comment by jderusse** [comment]
PR's ready (failing test is related to Documentation)

**Comment by dbu** [comment]
great, thanks!

**Pull request merged**